### ISSUE TITLE
Flag unused imports even with redefinition

### ISSUE THREAD

**Issue by charliermarsh** [charliermarsh/ruff#300]
Given this code:

```py
import os
import os
```

We return:

```
Found 1 error(s).
foo.py:2:1: F401 `os` imported but unused
```

Flake8 does this:

```
foo.py:2:1: F811 redefinition of unused 'os' from line 1
foo.py:2:1: F401 'os' imported but unused
```

My preference would be:

```
Found 2 error(s).
foo.py:1:1: F401 `os` imported but unused
foo.py:2:1: F401 `os` imported but unused
```

**Comment by Cxxshyy** [comment]
hi im fairly new to rust but can have a go at this so can you first tell me what file handle this and ill take a look

**Comment by charliermarsh** [comment]
@Cxxshyy - Awesome! I marked this as `good first issue` a few weeks back but I think it may be deceptively tricky. (It requires some extra book-keeping to understand when an import "redefines" another import, since you can have imports split across `if`-`else` blocks and so on.)

If you're looking for a place to contribute, a good starting point might be to take a rule you like from [`flake8-bugbear`](https://github.com/PyCQA/flake8-bugbear) and add it to Ruff (#389). I wrote some instructions on a similar issue in [here](https://github.com/charliermarsh/ruff/issues/312#issuecomment-1280008634) that would be helpful for understanding how to add a new check. Does that look interesting to you?

**Comment by Cxxshyy** [comment]
yes that looks interesting ill have a look at this tomorrow

**Comment by charliermarsh** [comment]
Cool, don't hesitate to ask questions, I'd love to be helpful.

**Comment by Cxxshyy** [comment]
will do thanks

**Comment by charliermarsh** [comment]
This got resolved when we added the redefined-while-unused checks.
